### ISSUE TITLE
Fix Multishipping dependency when replaced

### ISSUE THREAD

**Issue by JeroenVanLeusden** [Adyen/adyen-magento2#1292]
**Describe the bug**
You added Multishipping support in the latest release which is virtually never used. If removed from the codebase, DI compilation is broken.

```bash
Compilation was started.
Repositories code generation... 1/9 [===>------------------------]  11% < 1 sec 155.0 MiBPHP Fatal error:  Uncaught Error: Class 'Magento\Multishipping\Block\Checkout\Success' not found in /Users/x/Sites/x/vendor/adyen/module-payment/Block/Checkout/Multishipping/Success.php:39
Stack trace:
#0 /Users/x/Sites/x/vendor/composer/ClassLoader.php(571): include()
#1 /Users/x/Sites/x/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile('/Users/x/S...')
#2 [internal function]: Composer\Autoload\ClassLoader->loadClass('Adyen\\Payment\\B...')
#3 [internal function]: spl_autoload_call('Adyen\\Payment\\B...')
#4 /Users/x/Sites/x/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(134): class_exists('Adyen\\Payment\\B...')
#5 /Users/x/Sites/x/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(117): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->includeClass('Adyen\\Payment\\B...', '/Users/x/S...')
#6 /Users/x/Sites/x/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(87 in /Users/x/Sites/x/vendor/adyen/module-payment/Block/Checkout/Multishipping/Success.php on line 39
```

**To Reproduce**
1. Install magento, adyen v8 and replace multi-shipping module
```json
"replace": {
    "magento/module-multishipping": "*"
},
```
2. run `bin/magento setup:di:compile`

**Expected behavior**
Command completes without error.

**Magento version**
2.4.2

**Plugin version**
8.0.0

**Comment by tnaber** [comment]
Hi @JeroenVanLeusden,

Thank you for opening this issue. As indicated by the error, our Succes.php class extends Magento's default multishipping Succes.php class. The class is used when doing a multishipping checkout. Can you further elaborate why you think it is never used in practice?

Cheers,
Titus

**Comment by JeroenVanLeusden** [comment]
@tnaber Most people don't use the multishipping checkout and simply replace the module to prevent installation. If you did you cannot run `bin/magento setup:di:compile`.

If you want to keep this integration, a correct dependency should be added to the composer.json file or maybe migrate it to a separate package.

**Comment by norgeindian** [comment]
@tnaber , we have the same issue. As we do not use Multishipping, we removed the complete module from the code base. As no dependency is defined, this breaks the whole system when running `setup:di:compile`, like @JeroenVanLeusden  wrote.
So to use Adyen at all, we would need to include Multishipping again, which is pretty unsatisfying, as we don't need it.

**Comment by sprankhub** [comment]
The best way would probably be to have a separate `adyen-magento2-multishipping` package, which is optional and adds support for multishipping.

**Comment by JeroenVanLeusden** [comment]
As workaround we remove the files after `composer install`

```json
"scripts": {
    "post-install-cmd": [
      "rm -f vendor/adyen/module-payment/view/frontend/layout/multishipping_checkout_success.xml",
      "rm -f vendor/adyen/module-payment/view/frontend/layout/multishipping_checkout_billing.xml",
      "rm -rf vendor/adyen/module-payment/Block/Checkout/Multishipping"
    ]
  }
```

Agree with @sprankhub, it's probably best to have a separate package for multi-shipping support.

**Comment by norgeindian** [comment]
@tnaber , any news regarding this issue?

**Comment by tnaber** [comment]
Hi @norgeindian,

We discussed this internally, introducing a separate package for multishipping would be a large project and we are currently not able to provide the focus on that. Thus, for now, we have decided to add Magento_Multishipping to the dependencies in order to prevent the error. This means our plugin will not be able to compile without Magento_Multishipping installed, of course it is possible to disable this module in the settings.

If you would like to have a clean codebase I would recommend using the workaround proposed by @JeroenVanLeusden.

Cheers,
Titus
